test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -48,113 +54,178 @@ struct value_t {
+    get_record(const char *key)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        return (_collections.at(collection_name).values.at(key));
+        return (values.at(key));
     }
 
     void
-    insert_record(const std::string &collection_name, const char *key, const char *value)
+    insert_record(const char *key, const char *value)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.keys[key].exists = true;
+        keys[key].exists = true;
         value_t v;
         v.value = key_value_t(value);
-        c.values.emplace(key_value_t(key), v);
+        values.emplace(key_value_t(key), v);
     }
 
     void
-    update_record(const std::string &collection_name, const char *key, const char *value)
+    update_record(const char *key, const char *value)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.values.at(key).value = key_value_t(value);
+        values.at(key).value = key_value_t(value);
     }
 
     void
-    delete_record(const std::string &collection_name, const char *key)
+    delete_record(const char *key)
     {
         std::lock_guard<std::mutex> lg(_mtx);
-        auto &c = _collections.at(collection_name);
-        c.keys.at(key).exists = false;
-        c.values.erase(key);
+        keys.at(key).exists = false;
     }
 
+    std::map<key_value_t, key_t> keys = {};
+    std::map<key_value_t, value_t> values = {};
+    const std::string name;
+    const uint64_t id;
+
     private:
-    /* Take a const id, not a reference as we're copying in an atomic. */
-    std::string
+    std::mutex _mtx;
+    uint64_t _key_count;
+};
+
+/* Representation of the collections in memory. */
+class database {
+    public:
+    /*
+     * Add a new collection, this will create the underlying collection in the database.
+     */
+    void
+    add_collection(uint64_t key_count = 0)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        if (_session == nullptr)
+            _session = connection_manager::instance().create_session();
+        uint64_t next_id = _next_collection_id++;
+        std::string collection_name = build_collection_name(next_id);
+        _collections.emplace(
+          next_id, std::make_shared<collection>(next_id, key_count, collection_name));
+        testutil_check(
+          _session->create(_session, collection_name.c_str(), DEFAULT_FRAMEWORK_SCHEMA));
+        _tracking->save_schema_operation(
+          tracking_operation::CREATE_COLLECTION, next_id, _tsm->get_next_ts());
+    }
+
+    /* Get a collection using the id of the collection. */
+    std::shared_ptr<collection>
+    get_collection(uint64_t id)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        const auto &it = _collections.find(id);
+        if (it == _collections.end())
+            return (nullptr);
+        /* Make a copy. */
+        return std::shared_ptr<collection>(it->second);
```

Comment:
What's the idea behind the shared pointer? It's not synchronizing access to the underlying collection structure if that's the intention.

Reply:
As discussed, let's remove the use of `shared_ptr` and add a `TODO` here to refactor this into two separate things (one for the key range and one for the workload validation).